app/Repositories/DefaultAnimeRepository.php

```diff
@@ -124,7 +127,17 @@ public function getAiredBetween(
 //        ]);
 
         /** @noinspection PhpParamsInspection */
-        $queryable = $this->queryable(true)->whereRaw([
+        $queryable = $this->queryable(true);
+
+        if (!is_null($season) && !is_null($year)) {
+            $premiered = ucfirst($season)." {$year}";
+            $queryable = $this->queryable()
+                ->where("premiered", null)
+                ->orWhere("premiered", $premiered);
+        }
+
+        $queryable = $this->queryable()
+            ->whereRaw([
```

Comment:
In this case I would rather ask the question
"does the premiered date take precedence at all times?"
if yes then that should be part of the query filter, instead of adding it based on some condition.

I've tried to look into it, and so far I've found that the "premiered" field takes precedence, so the query logic should be like "if premiered not null, filter by that, otherwise filter by aired.to from"

Reply:
@pushrbx I think we'd need to have both because anything other than TV types will return premiered as null. The `aired.to` after that catches the rest

Reply:
> anything other than TV types will return premiered as null
I wish I could pin this somewhere so I won't forget this "business logic".